hubploy/auth.py

```diff
@@ -222,26 +222,24 @@ def cluster_auth_aws(deployment, project, cluster, zone, service_key):
 
     This changes *global machine state* on what current kubernetes cluster is!
     """
+    print(deployment, project, cluster, zone, service_key)
     # Get credentials from standard location
     service_key_path = os.path.join(
         'deployments', deployment, 'secrets', service_key
     )
 
-    original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
+    # if not running on CI, this is necessary to bypass local aws cli config
+    tmp_env = os.environ.copy()
```

Comment:
I'm slightly confused, since this config already existed under the `try`. Is what was needed unsetting `AWS_DEFAULT_PROFILE`?

Reply:
> Is what was needed unsetting AWS_DEFAULT_PROFILE? 

yes. for example if `aws-eks-config.cfg` contains:
```
[default]
aws_access_key_id = XXXXXXXXXXX
aws_secret_access_key = XXXXXXXXXX
```

But you have AWS_DEFAULT_PROFILE=scottyhq on the machine running hubploy, authentication fails

Reply:
Should I include changes to auth.py in this PR? I could refactor to fit the existing pattern:
`original_default_profile = os.environ.get("AWS_DEFAULT_PROFILE", None)` etc., but I thought the approach of passing `env` to subprocess calls was a bit cleaner?

Reply:
Yeah, passing to subprocess is cleaner! Would be great to have a separate PR that does that for all the env vars though.